src/app/tip.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { Tip } from './tip';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TipService {
+  tipURL = 'https://api.squiggle.com.au/?q=tips;year=2019;source=1';
+
+  constructor(private http: HttpClient) { }
+
+  getTips(): Observable<Tip[]> {
```

Comment:
```suggestion
  getTips(): Observable<Tip[]> {
    return this.http.get(this.tipURL)
    .pipe<Tip[]>(map((data: any) => data.tips));
  }
```
Don't you need to declare each property of the object Tip, like in the attached screenshot, or does `.pipe<tip[]>(map((data: any) => data.tips));` automatically take care of that?
<img width="746" alt="Screen Shot 2020-06-05 at 12 25 30 pm" src="https://user-images.githubusercontent.com/65341951/83830162-ce01a980-a727-11ea-894f-daa7cf65e665.png">

Reply:
Technically you don't, the `<Tip[]>` does the conversion, as long as the keys in the tip object matches the ones declared in the class.